docs/advanced_usage/api_lifecycle.rst

```diff
@@ -0,0 +1,98 @@
+.. _api_lifecycle:
+
+API Lifecycle
+=============
+
+To ensure a set of stable APIs, we introduced an API lifecycle.
+This will allow us to introduce/incubate new features while keeping
+the stable one up and running.
+
+Principle to work for
+---------------------
+
+If we extend our existing functionalities while following the
+`open-closed <https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle>`_
+principle, there should not be any breaking changes visible to our users.
+
+But as you all know, it is not that simple. Ensuring backward compatibilities is
+directly impacting **maintainability** in the following ways:
+
+* code we do not delete needs to be maintained (unittests and doc included)
+* legacy APIs may include switch cases in new APIs
+* legacy APIs may include dependencies to classes or modules we do not need
+
+
+Different use-cases
+-------------------
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Use-case
+     - User expectations
+     - Size of the complexity
+   * - Refinement of a functionality
+     - stable APIs
+     - medium
+   * - Refactoring activities of a functionality
+     - stable APIs
+     - medium
+   * - Incubation of a new functionality
+     - unstable APIs
+     - big
+
+API States
+----------
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - State
+     - Description
+   * - stable
+     - The API is stable and will not change in the future
+   * - incubation
+     - The API is not stable and may change in the future
+   * - deprecated
```

Comment:
Currently its missing how we want to handle deprecated functions right ? 
I guess add some warnings and put the new alternative function to incubation ??